### DISCUSSION TITLE
Dialog  Vue 3

### DISCUSSION THREAD

**Discussion by donPuerto** [quasarframework/quasar#10413] [General - Components / Directives / etc]
How to implement Dialog as a Custom Component?

**Comment by MilosPaunovic** [comment]
Start by looking at the offical docs here:
https://quasar.dev/vue-components/dialog

After which you'll have to try for yourself first and if you encounter a concrete problem, create a code sample of some kind so someone could help you.

**Reply by donPuerto** [reply]
@MilosPaunovic 
Im confuse with this template
How to trigger on and off here?
Do you have a complete code here?
```js
<template>
  <!-- notice dialogRef here -->
  <q-dialog ref="dialogRef" @hide="onDialogHide">
    <q-card class="q-dialog-plugin">
      <!--
        ...content
        ... use q-card-section for it?
      -->

      <!-- buttons example -->
      <q-card-actions align="right">
        <q-btn color="primary" label="OK" @click="onOKClick" />
        <q-btn color="primary" label="Cancel" @click="onCancelClick" />
      </q-card-actions>
    </q-card>
  </q-dialog>
</template>
```

**Reply by MilosPaunovic** [reply]
I'm not the best person to consult about Vue3 stuff, so you're gonna have to wait for someone else to intervene. In the meantime, take a deeper look at the official Quasar docs, you might find it helpful.

**Reply by ontwikkelfabriek** [reply]
@donPuerto 

You're using Quasar dialog **component**. Instead you have to use Quasar Dialog **plugin** to be able to create a Dialog with a **custom** component:
https://quasar.dev/quasar-plugins/dialog#invoking-custom-component

Please read the docs first. It's all explained in the regular Dialog docs:
https://quasar.dev/vue-components/dialog

_Masterclass TIP_

_Rather than cluttering your .vue templates with QDialogs, it’s best if you write a component for your dialog and use the Dialog Plugin to invoke it from anywhere in your app._

**Comment by tomschulze** [comment]
hi there,
i am not sure, but i guess you want to know how to trigger the dialog from a vue component? a minimal code example would really help to find a solution (e.g. via the [quasar codesandbox template](https://codesandbox.io/s/optimistic-poitras-4t3m3))

assuming im on the right track, it's described on the [dialog plugin page](https://quasar.dev/quasar-plugins/dialog#invoking-custom-component) in the quasar docs. first you define the component as in the docs. then, e.g., in a `*.vue` file you can trigger it like below

```js
<template>
  <q-btn round color="primary" icon="edit" @click="triggerDialog" />
</template>

<script>
import { defineComponent } from 'vue'
import MyCustomDialog from 'components/MyCustomDialog.vue'

export default defineComponent({
  name: 'SamplePage',
  setup() {
    const triggerDialog = () => {
      $q.dialog({
        component: MyCustomDialog
      })
    }

    return { triggerDialog }
  }
})
</script>
```

**Reply by donPuerto** [reply]
@tomschulze  This is a very straightforward solution. Let me try it.

On any luck, Do you have any working code sandbox solution for Vue Composable API for the Dialog?

https://quasar.dev/vue-composables/use-dialog-plugin-component

**Reply by tomschulze** [reply]
nah, sorry i don't have a working example.

**Reply by donPuerto** [reply]
@tomschulze This one is really interesting to implement however the doc is lacking on how to able to trigger the component atleast we could see the parent and child interaction

![image](https://user-images.githubusercontent.com/9329005/129916136-e23c2af1-191c-4ab0-8f78-8f4d1559f1f7.png)

Here's also some example
https://www.gitmemory.com/issue/quasarframework/quasar/9645/859948840
